### PR TITLE
Add Clear Site Data.

### PR DESCRIPTION
**Pull request by evanstade**
Most of this text needs to be worked into various parts of the CSD spec.

***
<a href="https://pr-preview.s3.amazonaws.com/evanstade/storage-buckets/pull/59.html" title="Last updated on Jan 11, 2023, 12:25 AM UTC (0ae7ae5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/storage-buckets/59/f93e40a...evanstade:0ae7ae5.html" title="Last updated on Jan 11, 2023, 12:25 AM UTC (0ae7ae5)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -514,4 +514,47 @@ See [[FS#sandboxed-filesystem]].
...
 </aside>
 
+<h3 id="storage-bucket-clear-site-data">Clear Site Data integration</h3>
+
+Issue: Update [[clear-site-data#header]].
+
+: "<dfn grammar>`storage:bucket-name`</dfn>"
+
+  ::  If the type string starts with "`storage:`" then the remaining characters after the
+      `:` will be taken to refer to a specific [=storage bucket=] in |origin|.
```
Comment (ayuishii) [inline index.bs:524]: should this point to https://url.spec.whatwg.org/#concept-url-origin?

File: index.bs [hunk 2]
```diff
@@ -514,4 +514,47 @@ See [[FS#sandboxed-filesystem]].
...
+
+<div algorithm>
+
+To <dfn>parse a Clear-Site-Data header with buckets</dfn>, execute the following steps:
+
+1. For each |type| in |header|, execute the following steps:
+
+    1. If |type| does not [=string/start with=] `"storage:"`, abort these steps.
+
+    1. Let |bucket name| be the [=code-unit-substring-by-positions|code unit substring=] from 8 to to |end| of |type|.
```
Comment (ayuishii) [inline index.bs:536]: extra `to`

File: index.bs [hunk 3]
```diff
@@ -514,4 +514,47 @@ See [[FS#sandboxed-filesystem]].
...
+
+    1. If the result of [=validate a bucket name=] with |bucket name| is failure, then abort these steps.
+
+    1. Append a [=tuple=] consisting of (`"storage-bucket"`, |bucket name|) to |types|.
```
Comment (ayuishii) [inline index.bs:540]: Maybe `Let |types| be an empty list` at the beginning to define this `types`?

File: index.bs [hunk 4]
```diff
@@ -514,4 +514,47 @@ See [[FS#sandboxed-filesystem]].
...
+  ::  If the type string starts with "`storage:`" then the remaining characters after the
+      `:` will be taken to refer to a specific [=storage bucket=] in |origin|.
+
+Issue: Update [[clear-site-data#parsing]] with these partial algorithms.
+
+<div algorithm>
+
+To <dfn>parse a Clear-Site-Data header with buckets</dfn>, execute the following steps:
+
+1. For each |type| in |header|, execute the following steps:
```
Comment (ayuishii) [inline index.bs:532]: Should `|header|` be defined before this, maybe like CSD?
`Let |header| be the result of extract header list values...` [link](https://w3c.github.io/webappsec-clear-site-data/#parsing)

File: index.bs [hunk 5]
```diff
@@ -514,4 +514,47 @@ See [[FS#sandboxed-filesystem]].
...
+
+    1. Append a [=tuple=] consisting of (`"storage-bucket"`, |bucket name|) to |types|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>clear data with buckets</dfn>, execute the following steps:
+
+1. For each |type| in |types|, execute the following steps:
```
Comment (ayuishii) [inline index.bs:548]: Define `|types|`? Or maybe add something like `clear data with buckets given a list of types`?

File: index.bs (hunk 3 above)
Comment (evanstade) [reply index.bs:540]: These steps are meant to be inserted into the existing algorithm that defines |types|. I tried to make that more clear in the `Issue` notes. WDYT?

Comment (ayuishii) [reply index.bs:541]: Ah I see, thanks for explaining. Updates to `Issue` lgtm!

**Review by ayuishii — APPROVED** [review]

**Pull request merged**